src/string-class.js

```diff
@@ -0,0 +1,197 @@
+   */
+  words() {
+    const newValue = this.trimSpace().removeSpecialChars();
+    if(newValue === '') {
+      return [];
+    }
+    return newValue.split(/\s+/);
+  },
+
+  /**
+   * removeSpecialChars method is used to replace non-alphanumeric characters
+   * including _ that are alone and replaces it with empty space
+   * @return {string}
+   */
+  removeSpecialChars() {
+    return this.replace(/[^\w\s]|_/g, '');
+  },
+
+
+  /**
+   * wordCount method counts the number of words in a string
+   * @return {int}: number of words
+   */
+  wordCount() {
+    return this.words().length;
+  },
+
+  /**
+   * toCurrency method formats a number by putting commas in the right place
+   * @return {string}: string representing the currency format
+   */
+  toCurrency() {
+    const parts = this.split(/[.]/g),
+      length = parts[0].length;
+    let firstCommaPos = (length % 3 === 0) ? 3 : length % 3;
+    const re = new RegExp('^[0-9]{'+firstCommaPos+'}|[0-9]{3}', 'g');
+    parts[0] = parts[0].replace(re, (number) => {
+      return `${number},`;
+    })
+    .replace(/,$/, '');
+    
+    if(parts[1]) {
+      return parts[0]+'.'+parts[1];
+    }
+    return parts[0];
+  },
+
+  /**
+   * fromCurrency format converts a string in the currency format by
+   * removing commas
+   * @return {string}: representing the pure digit format
+   */
+  fromCurrency() {
+    return this.replace(/,+/g, '');
+  },
+
+  /**
+   * inverseCase inverts the case of a character, a lower case character
+   * becomes upper case
+   * @return {string}: a string with inverted case
+   */
+  inverseCase() {
+    return this.replace(/[a-zA-Z]/g, (match) => {
+      return /[A-Z]/.test(match) ? match.toLower() : match.toUpper();
+    });
+  },
+
+  /**
+   * alternatingCase method alternates the case of the character in a string
+   * @return {string}; with alternated cases e.g aNoNyMoUs
+   */
+  alternatingCase() {
+    return this.replace(/[A-Za-z]/g, (match, index) => {
+      return (index % 2 === 0) ? match.toLower() : match.toUpper();
+    });
+  },
+
+  /**
+   * getMiddle method returns the character or the two characters at the mid
+   * position in a string
+   * @return {string}; of the middle character or two characters in the string
+   */
+  getMiddle() {
+    const length = this.length;
+    if(length % 2 === 0) {
+      return this[(length / 2) - 1] + this[length / 2];
+    }
+    return this[parseInt(length / 2)];
+  },
+
+  /**
+    * numberWords returns the word equivalent for the digits
+    * @return {string}; word equivalent for every digit
+    */
+  numberWords() {
+    const digitInWords = {
+      0: 'zero',
+      1: 'one',
+      2: 'two',
+      3: 'three',
+      4: 'four',
+      5: 'five',
+      6: 'six',
+      7: 'seven',
+      8: 'eight',
+      9: 'nine'
+    };
+    return this.replace(/\d/g, (match, index) => {
+      if(index === 0) {
+        return digitInWords[match];
+      }
+      return ' ' + digitInWords[match];
```

Comment:
__Code Quality - Critical__
You want to make use of ES6 string interpolation as stated in our style guide. Use template syntax instead of using `+` to concatenate strings. Like so:

```javaScript
return ` ${digitInWords[match]}`;
```

Take a look at our Airbnb [style guide](https://github.com/airbnb/javascript#es6-template-literals) on template syntax.